include/deal.II/lac/trilinos_tpetra_precondition_muelu.templates.h

```diff
@@ -0,0 +1,268 @@
+#include <Teuchos_ParameterList.hpp>
+
+#include <string>
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+#  include <deal.II/lac/trilinos_tpetra_precondition.h>
+#  include <deal.II/lac/trilinos_tpetra_precondition.templates.h>
+
+#  ifdef DEAL_II_TRILINOS_WITH_TPETRA_MUELU
+#    include <MueLu_CreateTpetraPreconditioner.hpp>
+#  endif
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace LinearAlgebra
+{
+  namespace TpetraWrappers
+  {
+#  ifdef DEAL_II_TRILINOS_WITH_IFPACK2
+#    ifdef DEAL_II_TRILINOS_WITH_TPETRA_MUELU
+
+    /* ---------------------- PreconditionAMGMueLu ------------------------ */
+    template <typename Number, typename MemorySpace>
+    PreconditionAMGMueLu<Number, MemorySpace>::AdditionalData::AdditionalData(
+      const bool         elliptic,
+      const bool         symmetric,
+      const bool         w_cycle,
+      const double       aggregation_threshold,
+      const int          smoother_sweeps,
+      const int          smoother_overlap,
+      const bool         output_details,
+      const std::string &smoother_type,
+      const std::string &coarse_type)
+      : elliptic(elliptic)
+      , symmetric(symmetric)
+      , w_cycle(w_cycle)
+      , aggregation_threshold(aggregation_threshold)
+      , smoother_sweeps(smoother_sweeps)
+      , smoother_overlap(smoother_overlap)
+      , output_details(output_details)
+      , smoother_type(smoother_type)
+      , coarse_type(coarse_type)
+    {}
+
+
+
+    template <typename Number, typename MemorySpace>
+    void
+    PreconditionAMGMueLu<Number, MemorySpace>::initialize(
+      SparseMatrix<Number, MemorySpace> &A,
+      Teuchos::ParameterList            &parameters)
+    {
+      this->parameter_list.setParameters(parameters);
+      Teuchos::RCP<Tpetra::Operator<Number,
+                                    TpetraTypes::LO,
+                                    TpetraTypes::GO,
+                                    TpetraTypes::NodeType<MemorySpace>>>
+        op = A.trilinos_rcp();
+      Teuchos::RCP<MueLu::TpetraOperator<Number,
+                                         TpetraTypes::LO,
+                                         TpetraTypes::GO,
+                                         TpetraTypes::NodeType<MemorySpace>>>
+        precon             = MueLu::CreateTpetraPreconditioner(op, parameters);
+      this->preconditioner = precon;
+    }
+
+
+
+    template <typename Number, typename MemorySpace>
+    void
+    PreconditionAMGMueLu<Number, MemorySpace>::initialize(
+      SparseMatrix<Number, MemorySpace> &A,
+      const AdditionalData              &ad)
+    {
+      Teuchos::ParameterList parameter_list;
+
+      if (ad.elliptic)
+        {
+          parameter_list.set("multigrid algorithm", "sa");
+        }
+      else
+        {
+          parameter_list.set("multigrid algorithm", "unsmoothed");
+        }
+      parameter_list.set("problem: symmetric", ad.symmetric);
+      parameter_list.set("aggregation: type", "uncoupled");
```

Comment:
There is not choice to be made for MueLu. MueLu only offers uncoupled aggregation. Coupled aggregation has been removed (https://github.com/trilinos/Trilinos/pull/11687) from MueLu.

Reply:
Ah good to know, I'll delete that line then.
I went with the User's guide which is from 2023/02 so before that was removed.